server/request_handler.go

```diff
@@ -147,6 +147,11 @@ func (r *RpcRequestHandler) processRequest(client RPCProxyClient, jsonReq *types
 	}
 	// Handle single request
 	rpcReq := NewRpcRequest(r.logger, client, jsonReq, r.relaySigningKey, r.relayUrl, origin, referer, isWhitehatBundleCollection, whitehatBundleId, entry, urlParams, r.chainID, r.rpcCache)
+
+	if r.req.Header.Get("X-Flashbots-Signature") != "" {
```

Comment:
i don't love this (passing signature and body around as part of the request), but i can see why we'd do it this way -- to only verify the signature on the methods that need it. so all in all it's also okay for me.